lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -1242,6 +1334,109 @@ bool SwiftASTManipulator::AddExternalVariables(
+        swift::SourceLoc(),
+        name,
+        swift::SourceLoc(),
+        name,
+        m_inner_function_decl->getDeclContext());
+    auto static_type = variable.GetStaticType();
+    auto type = swift_ast_context->ReconstructType(static_type.GetMangledTypeName());
+    param->setInterfaceType(type);
+    param->setSpecifier(swift::ParamSpecifier::Default);
+    param_decls.push_back(param);
+
+    auto resolved_type = swift_ast_context->ReconstructType(variable.GetType().GetMangledTypeName());
+    decl_refs.push_back(new (ast_context)
+                            swift::DeclRefExpr(variable.GetDecl(),
+                                               swift::DeclNameLoc(l_paren_loc),
+                                               true,
+                                               swift::AccessSemantics::Ordinary,
+                                               resolved_type));
+    names.push_back(variable.GetName());
+
+
+    if (swift::GenericTypeParamType *generic_type = llvm::dyn_cast<swift::GenericTypeParamType>(type)) {
+      auto parameter = new (ast_context) swift::GenericTypeParamDecl(m_function_decl,
+                                                                     ast_context.getIdentifier("T" + std::to_string(generic_param_index)),
+                                                                     swift::SourceLoc(),
+                                                                     0,
+                                                                     generic_param_index);
+      generic_param_index++;
+      generic_param_decls.push_back(parameter);
+      auto interface_type = parameter->getInterfaceType()->getMetatypeInstanceType().getPointer();
+      if (auto new_generic = llvm::dyn_cast<swift::GenericTypeParamType>(interface_type)) {
+        generic_param_types.push_back(new_generic);
+      }
+    }
+  }
+  auto inner_parameters = swift::ParameterList::create(ast_context,
+                                                       l_paren_loc,
+                                                       param_decls,
+                                                       r_paren_loc);
+  m_inner_function_decl->setName(swift::DeclName(ast_context,
+                                                 m_inner_function_decl->getBaseName(),
+                                                 inner_parameters));
+  m_inner_function_decl->setParameters(inner_parameters);
+
+  auto signature = swift::GenericSignature::get(
+      generic_param_types, swift::ArrayRef<swift::Requirement>());
+  m_inner_function_decl->setGenericSignature(signature);
+
+  auto generic_param_list = swift::GenericParamList::create(ast_context,
+                                                            l_paren_loc,
+                                                            generic_param_decls,
+                                                            r_paren_loc);
+  m_inner_function_decl->setGenericParams(generic_param_list);
```

Comment:
I'm a little lost, but: is m_inner_function_decl "__lldb_inner_exp"? Would it be easier to synthesize the AST using just the C++ APIs, instead of parsing a skeleton decl from source and then filling it in?

Reply:
It might be! I was mimicking what I saw was already there.

Reply:
Although now that I think about it, it might be harder, since we insert the user's expression inside `lldb_inner_expr`. Unless there is an easy way to parse the user's expression and insert the result inside the AST

Reply:
No, that's a good point.